### ISSUE TITLE
GLTFExporter: Add support for KHR_mesh_quantization.

### ISSUE THREAD

**Issue by xx0618** [mrdoob/three.js#20474]
In the past, my model was compressed with draco, and it was displayed on the web page normally, and it was also normal to export the file with GLTFExporter. However, due to the draco compressed model, the decompression speed is relatively slow, and the display is no problem if it is changed to GLTFPACK compression. But when I want to export the page, I am prompted that the bufferAttribute component type is not supported.

```js
var loader = new THREE.GLTFLoader();
loader.setMeshoptDecoder(MeshoptDecoder);
loader.load(model_path, function(gltf) {
theModel = gltf.scene;
theModel.position.y = -1.2;
modelg.add(theModel);
});
var urlsave = ‘http://192.168.101.113:8080/customProduct/saveCustomProduct’;
var qhxhr = new XMLHttpRequest();
qhxhr.open(‘POST’, urlsave, true);
var blobfile = null;
var fdata = new FormData();

var btnsave = document.getElementById(‘savejm’);
btnsave.addEventListener(‘click’, saves);
function saves() {
var exporter3D = new THREE.GLTFExporter();
exporter3D.parse(modelg, function(glbfile) {
if (glbfile instanceof ArrayBuffer) {
saveArrayBuffer(glbfile, ‘customize1.glb’);
} else {
var scenejson = JSON.stringify(glbfile, null, 2);
saveString(scenejson, ‘customize.glb’);
};
}, binary = true, onlyVisible = true );
};
function saveString(text, filename) {

  save(new Blob([text], {
  	type: 'text/plain'
  }), filename);
};
function saveArrayBuffer(buffer, filename) {
save(new Blob([buffer], {
type: ‘application/octet-stream’
}), filename);
};
function save(blob, filename) {
fdata.append(‘file’, blob, ‘customize.glb’);
qhxhr.upload.addEventListener(“progress”, function(result) {
if (result.lengthComputable) {
var percent = (result.loaded / result.total * 100).toFixed(2);
console.log(percent);
}
}, false);
qhxhr.send(fdata);
console.log(“ff”);
};
```

Hope to support bufferAttribute component type in GLTFExporter.

![微信截图_20201008112427](https://user-images.githubusercontent.com/71865768/95420116-85742a00-096d-11eb-93fc-d42892a91c12.png)

**Comment by Mugen87** [comment]
From the forum: https://discourse.threejs.org/t/gltfexporter-unsupported-bufferattribute-component-type/19452

**Comment by Mugen87** [comment]
@xx0618 Please also share a problematic glTF model for testing as requests by @donmccurdy here: 

https://discourse.threejs.org/t/gltfexporter-unsupported-bufferattribute-component-type/19452/4

**Comment by donmccurdy** [comment]
I suspect it's any model using `KHR_mesh_quantization` that will cause this error. If so, this file should reproduce the issue:

[glTF-Quantized 2.zip](https://github.com/mrdoob/three.js/files/5361574/glTF-Quantized.2.zip)

**Comment by xx0618** [comment]
> 我怀疑使用的任何模型`KHR_mesh_quantization`都会导致此错误。如果是这样，此文件应重现该问题：
> 
> [glTF量化2.zip](https://github.com/mrdoob/three.js/files/5361574/glTF-Quantized.2.zip)

Yes, the problem occurs again. GLTFExporter cannot save the page.
![微信截图_20201012100015](https://user-images.githubusercontent.com/71865768/95697691-231f7000-0c72-11eb-8051-f811c814f94e.png)

**Comment by Mugen87** [comment]
Small update on this issue: When fixing a bug in context of interleaved data, I've also made a test with `glTF-Quantized 2.zip` since it contains interleaved data.

I've realized it will be a bit complicated to add support for `KHR_mesh_quantization` since the exporter actually uses buffer data at various places. At least when recomputing normals, data of typed arrays which are not `Float32Array` have to be decoded into floats and then decoded back after the computation.

Besides, it's necessary to honor specific (byte) strides and padding values in order to write valid buffer data for quantized values.
